test/unittest/models/models_test_utils.cpp

```diff
@@ -376,7 +376,7 @@ void GraphWatcher::compareFor(const std::string &reference,
 
     auto it = nodes.begin();
     for (; it != nodes.end() - 1; ++it) {
-      it->forward(iteration, !(it + 1)->supportInPlace());
+      it->forward(iteration, it->isOutputNode() | !optimize);
```

Comment:
was this the reason for test fail? :p

Reply:
yes :)

Reply:
with these optimizations, we cannot always match all the outputs as they get overwritten before they can be verified.